src/inference/src/ie_core.cpp

```diff
@@ -245,32 +245,74 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             std::shared_ptr<ie::ICacheManager> _cacheManager;
         };
 
-        bool flag_allow_auto_batching = true;
+        // Core default supported properties and values, which are core global properties.
+        // They don't require a specified device.
+        const ov::AnyMap default_core_supported_properties = {ov::cache_dir(""),
+                                                              ov::hint::allow_auto_batching(true),
+                                                              ov::auto_batch_timeout(1000),
+                                                              ov::force_tbb_terminate(false)};
+
+        void set_core_config(ov::AnyMap& config) {
+            for (auto it = config.begin(); it != config.end();) {
+                auto item = default_core_supported_properties.find(it->first);
+                if (item == default_core_supported_properties.end()) {
+                    it++;
```

Comment:
Here should report exception if the input config name is not in `default_core_supported_properties`

Reply:
I suppose wee should not. Originally, `setAndUpdate` just removes core config keys from `config`. If some keys in `config` are not core-config, they just ignored.

Reply:
can we add detailed documentation to methods to describe what they do? this file is getting more and more complex and we need to start documenting it.

Reply:
Originally, I thought if non core-configs were passed by core.set_property() it should throw exception to warn caller, but there was a CI error due to one test case for PERF_COUNT property is set by core.set_property() without device name. Anyhow, ignore the non core-configs is ok.
yes, I will add details documentation here.